world/clientlist.cpp

```diff
@@ -94,14 +94,17 @@ void ClientList::GetCLEIP(uint32 in_ip) {
 	int count = 0;
 	iterator.Reset();
 
+	const auto& zones = Strings::Split(RuleS(World, IPExemptionZones), ",");
+
 	while (iterator.MoreElements()) {
 		cle = iterator.GetData();
-		
-		const auto zones = Strings::Split(RuleS(World, IPExemptionZones), ",");
-		for (const auto &z : zones) {
-			if (Strings::ToUnsignedInt(z) == cle->zone()) {
-				iterator.Advance();
-				continue;
+
+		if (!zones.empty() && cle->zone()) {
+			for (const auto& z : zones) {
+				if (Strings::ToUnsignedInt(z) == cle->zone()) {
+					iterator.Advance();
+					continue;
```

Comment:
The `Advance()` and `continue` are still within the inner `for (const auto& z : zones)` loop. These should only be called once after the loop if a match was found to continue the main `while (iterator.MoreElements())` loop. This would also lead to crashes again if the same zone was listed twice in the rule and matched cle->zone.

It should be breaking from the zones loop when a zone match is found and then only calling advance + continue outside it. I'd suggest something like:

```cpp
auto it = std::ranges::find_if(zones,
		[cle](const auto& z) { return Strings::ToUnsignedInt(z) == cle->zone(); });

if (it != zones.end())
{
		iterator.Advance();
		continue;
}
```

Reply:
Hopefully what I just changed is what you mean.